contracts/CarFactoryAndWorker/CarFactory.tsol

```diff
@@ -39,14 +49,25 @@ contract CarFactory {
         });
     }
 
-    function getAddress(address _accountAddress) public returns(address) {
-        tvm.accept();
+    function getAddress(address _accountAddress) public view returns(address) {
         TvmCell accountInitState = buildInitParams(_accountAddress);
-        emit AddressReturn(address(tvm.hash(accountInitState)));
         return address(tvm.hash(accountInitState));
     }
 
-    //function to senf offer to the worker.
+    function deployNewContract(address _ownerAddress) internal {
+        TvmCell stateInit = buildInitParams(_ownerAddress);
+
+        address accountAddress = new Workers{
+                stateInit: stateInit,
+                code: workerContractCode,
+                value: 1 ever
+
+        }(workerContractCode);
+        emit AccountDeployed(accountAddress, _ownerAddress);
+    }
+
+
+    //function to send offer to the worker.
     function sendOffer(string offer, address _destAddress) public {
```

Comment:
You mustn't use pre-calculated addresses, this method should receive workerOwnerAddress